doc/rdoc/markup_reference.rb

```diff
@@ -7,72 +7,74 @@
+# The sources of \RDoc documentation vary according to the type of file:
 #
-# The comment associated with
-# a Ruby class, module, method, alias, constant, or attribute
-# becomes the documentation for that defined object:
+# - <tt>.rb</tt> (Ruby code file):
 #
-# - In a Ruby file, that comment immediately precedes
-#   the definition of the object.
-# - In a C file, that comment immediately precedes
-#   the function that implements a method,
-#   or otherwise immediately precedes the definition of the object.
+#   - Markup may be found in Ruby comments:
+#     A comment that immediately precedes the definition
+#     of a Ruby class, module, method, alias, constant, or attribute
+#     becomes the documentation for that defined object.
+#   - An \RDoc directive may be found in:
 #
-# In either a Ruby or a C file,
-# \RDoc ignores comments that do not precede object definitions.
+#     - A trailing comment (on the same line as code);
+#       see <tt>:nodoc:</tt>, <tt>:doc:</tt>, and <tt>:notnew:</tt>.
+#     - A single-line comment;
+#       see other {Directives}[rdoc-ref:RDoc::MarkupReference@Directives].
 #
-# In an \RDoc file, the text is not associated with any code object,
-# but may (depending on how the documentation is built),
-# become a separate page.
+#   - Documentation may be derived from the Ruby code itself;
+#     see {Documentation Derived from Ruby Code}[rdoc-ref:RDoc::MarkupReference@Documentation+Derived+from+Ruby+Code].
 #
-# Almost all examples on this page are all RDoc-like;
-# that is, they have no comment markers like Ruby <tt>#</tt>
-# or C <tt>/* ... */</tt>.
+# - <tt>.c</tt> (C code file): markup is parsed from C comments.
+#   A comment that immediately precedes
+#   a function that implements a Ruby method,
+#   or otherwise immediately precedes the definition of a Ruby object,
+#   becomes the documentation for that object.
+# - <tt>.rdoc</tt> (\RDoc markup text file) or <tt>.md</tt> (\RDoc markdown text file):
+#   markup is parsed from the entire file.
+#   The text is not associated with any code object,
+#   but may (depending on how the documentation is built),
+#   become a separate page.
+#
+# An <i>RDoc segment</i> (hereafter simply called a _segment_) is:
+#
+# - A (possibly multi-line) comment in a Ruby or C file
+#   that generates \RDoc documentation (as above).
+# - The entire markup (<tt>.rdoc</tt>) file or markdown (<tt>.md</tt>) file,
+#   (which is usually multi-line).
 #
 # === Margins
 #
-# In a multi-line comment,
-# \RDoc looks for the comment's natural left margin,
-# which becomes the <em>base margin</em> for the comment
-# and is the initial <em>current margin</em> for the comment.
+# In a multi-line segment,
+# \RDoc looks for the segment's natural left margin,
+# which becomes the <em>base margin</em> for the segment
+# and is the initial <em>current margin</em> for the segment.
 #
 # The current margin can change, and does so, for example in a list.
```

Comment:
Would it make sense to talk about margins in the context of blocks since each block has the same margin?